plumbing/format/commitgraph/encoder.go

```diff
@@ -56,9 +56,6 @@ func (e *Encoder) Encode(idx Index) error {
 			return err
 		}
 	}
-	if err != nil {
-		return err
-	}

```

Comment:
Can you explain me why this is not necessary? It could be rewritten to `else` but the error handling should stay there.

Reply:
Because there's no way for `err` to not be `nil`. It's only assigned as `var err error` above, everything else below uses err in its own `if` scope and deals with it directly.

Reply:
@filipnavara I assume that means you actually want to handle any error returned from `encodeCommitData`? Happy to fix that, but currently it's not going to happen, because the outer `err` gets shadowed.

Reply:
Yep, the shadowing was unintentional.

Reply:
Fixed and pushed!

Reply:
Thanks